cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,415 @@
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('addExistsFilterButton').click();
+      });
+      // Verify full pill text
+      // globalFilterLabelValue gives the inner element, but we may want all the text in the filter pill
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
+      })
+        .parent()
+        .should('have.text', filterFieldText + ': ' + 'exists');
+      cy.getElementByTestId('discoverQueryHits').should(
+        'have.text',
+        expectedQueryHitsAfterFilterApplied
+      );
+    }
+  );
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Sends a new query via the query multiline editor.
+ * @param del true/false. true: Deletes character to the right of the cursor; false: Deletes character to the left of the cursor
+ * @see https://docs.cypress.io/api/commands/type#Arguments
+ */
+export function sendQueryOnMultilineEditor(query) {
+  const getMultilineEditor = () => {
+    cy.getElementByTestId('headerGlobalNav').click(); // remove syntax helper
+    return cy.get('.view-line');
+  };
+  // Clear default text on the editor by an alternative method, since
+  // cy.clear() won't work for some reason
+  getMultilineEditor()
+    .invoke('text')
+    .then(function ($content) {
+      const contentLen = $content.length + 1;
+      getMultilineEditor().type('a'); // make sure we're at the end of the string
+      getMultilineEditor().type('{backspace}'.repeat(contentLen));
+    });
+  // Type query
+  getMultilineEditor().type(query);
+  // Send query
+  cy.getElementByTestId('querySubmitButton').click();
+}
+
+/**
+ * Set the query editor language
+ * @param language Accepted values: 'DQL', 'Lucene', 'OpenSearch SQL', 'PPL'
+ */
+export function setQueryEditorLanguage(language) {
+  cy.getElementByTestId('headerGlobalNav').click(); // remove helper message
+  cy.getElementByTestId('queryEditorLanguageSelector').click();
+  cy.getElementByTestId('queryEditorLanguageOptions').find('button').contains(language).click();
+}
+
+/**
+ * Click on the sidebar collapse button.
+ * @param {boolean} collapse true for collapsing, false for expanding
+ */
+export function clickSidebarCollapseBtn(collapse = true) {
+  if (collapse) {
+    cy.getElementByTestId('euiResizableButton').trigger('mouseover').click();
+    cy.get('.euiResizableToggleButton').click({ force: true });
+  } else {
+    cy.get('.euiResizableToggleButton').click();
+  }
+}
+
+/**
+ * Check the results of the sidebar filter bar search.
+ * @param {string} search text to look up
+ * @param {string} assertion the type of assertion that is going to be performed. Example: 'eq', 'include'. If an assertion is not passed, a negative test is performend.
+ */
+export function checkSidebarFilterBarResults(search, assertion) {
```

Comment:
For this and other functions (as well as callback functions), please see: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9119#discussion_r1898721855